src/MOIWrapper.jl

```diff
@@ -0,0 +1,345 @@
+        end
+
+        ci = MOI.get(userOptimizer, MOI.ListOfConstraintIndices{F,S}())
+
+        if S == MOI.ZeroOne
+            updateZeroOneIndices(userOptimizer, mapping, ci, zeroOneIndices)
+        elseif S == MOI.Integer
+            updateIntegerIndices(userOptimizer, mapping, ci, integerIndices)
+        end
+
+        ## Update conmap for (F,S) for F != MOI.SingleVariable
+        ## Single variables are treated by bounds in Cbc, so no
+        ## need to add a row
+        if F != MOI.SingleVariable
+            for i in 1:length(ci)
+                mapping.conmap[ci[i]] = MOI.ConstraintIndex{F,S}(nbRows + i)
+            end
+            nbRows += MOI.get(userOptimizer, MOI.NumberOfConstraints{F,S}())
+        end
+    end
+
+    cbcModelFormat = CbcModelFormat(nbRows, nbCols)
+
+    copyconstraints!(cbcModelFormat, userOptimizer, mapping)
+    updateBoundsForBinaryVars!(cbcModelFormat.col_lb, cbcModelFormat.col_ub, zeroOneIndices)
+
+
+    ## Copy objective function
+    objF = MOI.get(userOptimizer, MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}())
+    loadObj(cbcModelFormat, mapping, objF)
+    sense = MOI.get(userOptimizer, MOI.ObjectiveSense())
+    MOI.set!(cbcOptimizer, MOI.ObjectiveSense(), sense)
+
+    ## Load the problem to Cbc
+    CbcCI.loadProblem(cbcOptimizer.inner, cbcModelFormat.constraint_matrix,
+    cbcModelFormat.col_lb, cbcModelFormat.col_ub, cbcModelFormat.obj,
+    cbcModelFormat.row_lb, cbcModelFormat.row_ub)
+
+    ## Set integer variables
+    for colIdx in vcat(integerIndices, zeroOneIndices)
+        CbcCI.setInteger(cbcOptimizer.inner, colIdx-1)
+    end
+
+    return MOI.CopyResult(MOI.CopySuccess, "Model was copied succefully.", MOIU.IndexMap(mapping.varmap, mapping.conmap))
+end
+
+
+function MOI.optimize!(cbcOptimizer::CbcOptimizer)
+    # Call solve function
+    CbcCI.solve(cbcOptimizer.inner)
+end
+
+
+
+## canadd, canset, canget functions
+
+function MOI.canaddvariable(cbcOptimizer::CbcOptimizer)
+    return false
+end
+
+## supports constraints
+
+
+MOI.supportsconstraint(::CbcOptimizer, ::Type{<:Union{MOI.ScalarAffineFunction{Float64}, MOI.SingleVariable}},
+::Type{<:Union{MOI.EqualTo{Float64}, MOI.Interval{Float64}, MOI.LessThan{Float64},
+MOI.GreaterThan{Float64}, MOI.ZeroOne, MOI.Integer}}) = true
+
+MOI.supports(cbcOptimizer::CbcOptimizer, object::MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}) = true
+
+## Set functions
+
+function MOI.write(cbcOptimizer::CbcOptimizer, filename::String)
+    writeMps(cbcOptimizer.inner, filename)
+end
+
+
+# empty!
+function MOI.empty!(cbcOptimizer::CbcOptimizer)
+    cbcOptimizer.inner = CbcModel()
+end
+
+
+function MOI.set!(cbcOptimizer::CbcOptimizer, object::MOI.ObjectiveSense, sense::MOI.OptimizationSense)
+    if sense == MOI.MaxSense
+        CbcCI.setObjSense(cbcOptimizer.inner, -1)
+    end ## Other senses are set as minimization (cbc default)
```

Comment:
What happens this function is called with `MaxSense` and then `MinSense`? The `MinSense` will never be set.